modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/kill/LocalToClusterKillHandlerWrapper.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.exec.kill;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.ignite.internal.cluster.management.topology.api.LogicalTopologyService;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.network.MessagingService;
+import org.apache.ignite.internal.sql.engine.api.kill.CancellableOperationType;
+import org.apache.ignite.internal.sql.engine.api.kill.OperationKillHandler;
+import org.apache.ignite.internal.sql.engine.message.CancelOperationRequest;
+import org.apache.ignite.internal.sql.engine.message.CancelOperationResponse;
+import org.apache.ignite.internal.sql.engine.message.SqlQueryMessagesFactory;
+
+/**
+ * Wrapper for {@link OperationKillHandler} that calls a local handler on each node in the cluster.
+ */
+class LocalToClusterKillHandlerWrapper implements OperationKillHandler {
+    /** Messages factory. */
+    private static final SqlQueryMessagesFactory FACTORY = new SqlQueryMessagesFactory();
+
+    /** Logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(LocalToClusterKillHandlerWrapper.class);
+
+    /** Maximum time to wait for a remote response. */
+    private static final long RESPONSE_TIMEOUT_MS = TimeUnit.MINUTES.toMillis(5);
+
+    private final OperationKillHandler localHandler;
+    private final CancellableOperationType type;
+    private final String localNodeName;
+    private final LogicalTopologyService logicalTopologyService;
+    private final MessagingService messageService;
+
+    LocalToClusterKillHandlerWrapper(
+            OperationKillHandler localHandler,
+            CancellableOperationType type,
```

Comment:
I think, it's better to derive type from handler itself

Reply:
Missed it, fixed, thanks